zk-desktop.el

```diff
@@ -177,28 +234,26 @@ To quickly change this setting, call `zk-desktop-add-toggle'."
 
 ;;;###autoload
 (defun zk-desktop-select ()
-  "Select a ZK-Desktop to work with."
+  "Select a ZK-Desktop to work with.
+Return the buffer object visiting the selected or created
+desktop."
   (interactive)
   (unless zk-desktop-directory
     (error "Please set `zk-desktop-directory' first"))
   (let* ((last-command last-command)
          (desktop
           (completing-read "Select or Create ZK-Desktop: "
-                           (directory-files
-                            zk-desktop-directory
-                            nil
-                            (concat
-                             zk-desktop-basename
-                             ".*"))
-                           nil nil nil nil
-                           (concat zk-desktop-basename " ")))
+                           (directory-files zk-desktop-directory
+                                            nil
+                                            (concat
+                                             zk-desktop-basename
+                                             ".*"))
+                           nil nil zk-desktop-basename nil))
```

Comment:
From the doc-string for `completing-read`, regarding use of INITIAL-INPUT:
"This feature is deprecated--it is best to pass nil
  for INITIAL-INPUT and supply the default value DEF instead.  The
  user can yank the default value into the minibuffer easily using
  M-n."

Reply:
Yeah, I realize that about `completing-read`, but sometimes it just doesn't make sense. `zk-desktop-basename` is not the default value; it is a common portion of zk-desktop names that the user then adds to. INITIAL-INPUT seems like the most natural place to put that in, though perhaps we could add something to the prompt text instead? Like `"Select or Create ZK-Desktop (following [value of zk-desktop-basename]): ` but less awkward...